### ISSUE TITLE
Q: truncate (or drop and re-create) tables for testing

### ISSUE THREAD

**Issue by idc77** [ent/ent#2127]
I'm writing tests and I'd like to truncate tables (mysql this time around) before the tests are ran.

```go
client = enttest.Open(t, dialect.MySQL, config.MysqlDsnTest, migrate.WithDropColumn)
```

`Cannot use 'migrate.WithDropColumn' (type func(b bool) MigrateOption) as the type Option`

How can I drop and recreate or truncate tables with `enttest`, please?
v0.9.1, skimmed issues
Not too familiar with Go and SQL, been using mongodb mostly before, so there might be something obvious I'm missing.

**Comment by idc77** [comment]
I've done this
```go
	client = enttest.Open(
		t,
		dialect.MySQL,
		config.MysqlDsnTest,
		enttest.WithMigrateOptions(
			migrate.WithDropIndex(true),
			migrate.WithDropColumn(true),
		))
``` 
but it doesn't drop anything, or maybe it does but the records are still present in the tables.

**Comment by a8m** [comment]
Hey @idc77, you can do the same thing I do on [ent integration tests](https://github.com/ent/ent/blob/master/entc/integration/integration_test.go#L1824-L1839). I delete all entities in the tables using `client.T.Delete().Exec(ctx)`. Does it solve your issues? Thanks

**Comment by idc77** [comment]
Hi @a8m ,
thanks for response
<blockquote>
The workaround of deleting all records doesn't reset the id serial counter.
</blockquote>
In this test, for a url shortener, I rely on the counter to start from 1.
something like
`migrate.WithDropTable(true)`
would be a welcome addition.

I agree this might be an edge case, but the need to drop tables is not unheard of, be it spring boot/hibernate, gorm or doctrine they all have the option to drop tables before migration.
You're probably busy with other, more interesting things, so no rush. I can manually drop tables to run tests, would be nice to have this feature though.
Alright, thanks for feedback.

**Comment by Yiling-J** [comment]
@idc77 Usually in tests, you only need to truncate all tables after each test. **OR**, if you run tests parallel and access same table, wrap your test ent client in a transaction at the beginning and rollback the transaction after the test. GORM has `DropTable` API to allow you writing custom migrate scripts, but ent is scheme based and no such thing now.

**Comment by idc77** [comment]
@Yiling-J What is usual or not depends on the person you ask and the use case.
In this case, like I wrote, I rely on the autoincrement functionality of the database starting from 1. And that is the end of this discussion.

**Comment by HurSungYun** [comment]
I know it's kinda risky feature to support so I think it's not a `must have` feature, but it would be great to have something like `DropTable` API for testing. 

Sharing my use case here. `Test_CreateUser2` always fails because INNODB auto_increment ID have not reset when truncating. (So I drop table via raw query each time)

```go
...
type TestEnv struct {
	db             *ent.Client
}

func newTestEnv() *TestEnv {
	var (
		env = &TestEnv{}
		err error
	)
	cfg := tests.SetUpTestConfig()
	env.db, env.mysql, err = db.NewMySQL(cfg)
	if err != nil {
		log.Panic().Err(err).Send()
	}

	err = env.db.Schema.Create(
		context.Background(),
		migrate.WithForeignKeys(false), // Disable foreign keys.
	)
	if err != nil {
		log.Panic().Err(err).Send()
	}
	return env
}

func (e *TestEnv) cleanUp(a *assert.Assertions) {
	ctx := context.Background()
	if _, err := e.db.User.Delete().Exec(ctx); err != nil {
		a.FailNow(err.Error(), "failed to truncate user table")
	}
}

func Test_CreateUser(t *testing.T) {
	a := assert.New(t)

	env := newTestEnv()
	defer env.cleanUp(a)

	// User has auto_increment ID and Name
	record := User{
		Name: "a",
	}

	if _, err := CreateUser(c, record); err != nil {
		a.FailNow(err.Error(), "failed to setup test fixtures")
	}

	// auto_increment ID should be known to test its data is set
	user, err := GetUser(context.Background(), 1)
	if err != nil {
		a.FailNow(err.Error(), "failed to get user")
	}
	a.EqualValues(record.Name, user.Name)
}

func Test_CreateUser2(t *testing.T) {
	a := assert.New(t)

	env := newTestEnv()
	defer env.cleanUp(a)

	// User has auto_increment ID and Name
	record := User{
		Name: "b",
	}

	if _, err := CreateUser(c, record); err != nil {
		a.FailNow(err.Error(), "failed to setup test fixtures")
	}

	// auto_increment ID should be known to test its data is set
	user, err := GetUser(context.Background(), 1)
	if err != nil {
		// ALERT: Test fails here because auto_increment ID didn't reset
		a.FailNow(err.Error(), "failed to get user")
	}

	// not reached
	a.EqualValues(record.Name, user.Name)
}

```

Or, maybe selecting database when creating schema could be another solution. When people want to run tests in parallel, they tends to create schemas with random names.

```go

err = env.db.Schema.Create(
	context.Background(),
	migrate.WithDatabase("test_db"), // Select Database. e.g.) `SELECT "test_db"`
)
```

**Comment by Yiling-J** [comment]
@HurSungYun One thing I try to understand in your snippet, why can't you just get user ID from your `CreateUser`?
```go
if user, err := CreateUser(c, record); err != nil {
		a.FailNow(err.Error(), "failed to setup test fixtures")
	}

	// auto_increment ID should be known to test its data is set
	user, err := GetUser(context.Background(), user.ID)
	if err != nil {
		a.FailNow(err.Error(), "failed to get user")
	}
```

**Comment by HurSungYun** [comment]
@Yiling-J you are right. My bad. Thanks for inspecting my code :)
